pkg/cli/debug/debug.go

```diff
@@ -455,17 +456,24 @@ func (o *DebugOptions) RunDebug() error {
 	pod.Name, pod.Namespace = fmt.Sprintf("%s-debug-%s", generateapp.MakeSimpleName(infos[0].Name), utilrand.String(5)), ns
 	o.Attach.Pod = pod
 
+
 	if len(o.ContainerName) == 0 && len(pod.Spec.Containers) > 0 {
+		c, err := podcmd.FindOrDefaultContainerByName(pod, o.ContainerName, o.Quiet, o.ErrOut)
```

Comment:
@holzman thanks for spending effort on this PR. 

As stated in https://github.com/openshift/oc/blob/9f903f99a6650da83a0409f81aff3ced0acaad38/pkg/cli/debug/debug.go#L245, `oc debug` particularly defaults to first container and if user wants to use another pod, user can specify it via `container` flag. This change may break some scripts possibly relying on using the first container in pod rather than the one with annotation.

Reply:
Yes, but that behavior predates the default container annotations. I'd argue that ignoring the annotation is now a bug rather than a feature.